packages/cli/src/util/oauth.ts

```diff
@@ -398,3 +401,56 @@ function canParseURL(url: string) {
     return false;
   }
 }
+
+/**
+ * Perform Token Introspection Request.
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc7662#section-2.1
+ */
+export async function inspectTokenRequest(token: string): Promise<Response> {
+  return fetch((await as()).introspection_endpoint, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'user-agent': ua,
+    },
+    body: new URLSearchParams({ token }),
+  });
+}
+
+/**
+ * @see https://datatracker.ietf.org/doc/html/rfc7662#section-2.2 */
+interface AccessToken {
+  /**
+   * Integer timestamp, measured in the number of seconds
+   * since January 1 1970 UTC, indicating when this token will expire.
+   */
+  exp: number;
+  /** Whether or not the presented token is active. */
+  active: boolean;
+  token_type: 'access_token';
+  /** The authorizing principal's team. */
+  team_id?: string;
+  session_id: string;
+}
+
+/**
+ * Process Token Introspection Response.
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc7662#section-2.2
+ */
+export async function processInspectTokenResponse(
+  response: Response
+): Promise<[InspectionError] | [null, AccessToken]> {
+  try {
+    const token = await response.json();
+    if (!token || typeof token !== 'object' || !('session_id' in token)) {
+      throw new Error('Invalid token response');
+    }
+    return [null, token];
+  } catch (cause) {
+    return [new InspectionError('Could not inspect token.', { cause })];
```

Comment:
Nit: use "introspect" everywhere?

```suggestion
    return [new IntrospectionError('Could not introspect token.', { cause })];
```